### PR TITLE
Recalibration branch finished

### PR DESCRIPTION
**Pull request by YoungseogChung**
Added std optimization based recalibration, added prediction interval outputs, added recalibrated prediction interval outputs. 
Added tests and examples for new code.

### CODE REVIEW

File: tests/test_recalibration.py [hunk 1]
```diff
@@ -87,3 +92,119 @@ def test_miscal_area_recalibration_on_test_set(supply_test_set):
...
+    )
+    recal_miscal = miscalibration_area(y_pred, rms_cal_ratio * y_std, y_true)
+
+    assert np.abs(rms_cal_ratio - 0.34900989073212507) < 1e-6
+    assert np.abs(recal_ma_cal - 0.06945555555555555) < 1e-6
+    assert np.abs(recal_rms_cal - 0.08570902541177935) < 1e-6
+    assert np.abs(recal_miscal - 0.07011706864564003) < 1e-6
+
+
+def test_miscal_area_std_recalibration_on_test_set(supply_test_set):
```
Comment (IanChar) [inline tests/test_recalibration.py:149]: I would vote for having the function that each of these is testing directly in the name, which a lot of the tests in this file do not have. For example, I think this one should be test_optimize_recalibration_ratio_miscal_area_criterion().

File: examples/viz_recalibrate_readme.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
+"""
+Examples of code for recalibration.
```
Comment (IanChar) [inline examples/viz_recalibrate_readme.py:2]: Can we add onto this and say explicitly it produces the example in the README.

**Review by IanChar — CHANGES_REQUESTED** [review]

**Review by willieneis — DISMISSED** [review]
Looks good with a few minor adjustments!

File: uncertainty_toolbox/recalibration.py [hunk 3]
```diff
@@ -1,9 +1,12 @@
 """
 Recalibrating uncertainty estimates.
 """
-
+import tqdm
```
[the remark this answers is not in the document]
Comment (willieneis) [reply uncertainty_toolbox/recalibration.py:4]: updated!

File: uncertainty_toolbox/recalibration.py [hunk 4]
```diff
@@ -1,9 +1,12 @@
...
 import numpy as np
 from sklearn.isotonic import IsotonicRegression
+import uncertainty_toolbox as uct
+from scipy import special
```
[the remark this answers is not in the document]
Comment (willieneis) [reply uncertainty_toolbox/recalibration.py:7]: updated!

File: uncertainty_toolbox/recalibration.py [hunk 5]
```diff
@@ -101,3 +104,42 @@ def iso_recal(exp_props, obs_props):
         raise RuntimeError("Failed to fit isotonic regression model")
 
     return iso_model
+
+
+def optimize_recalibration_ratio(y_mean, y_std, y_true, criterion="ma_cal"):
```
[the remark this answers is not in the document]
Comment (willieneis) [reply uncertainty_toolbox/recalibration.py:109]: added!

File: examples/viz_recalibrate_readme.py (hunk 2 above)
Comment (willieneis) [reply examples/viz_recalibrate_readme.py:2]: added!

File: tests/test_recalibration.py (hunk 1 above)
Comment (willieneis) [reply tests/test_recalibration.py]: updated by ian!

**Review by willieneis — APPROVED** [review]
looks good!

**Pull request merged**